### DISCUSSION TITLE
Integrate the comminication of a new Battery Computer from BlueBattery

### DISCUSSION THREAD

**Discussion by Karatzie** [fl4p/batmon-ha#141] [Ideas]
Hi Fabian,
the company [BlueBattery](https://www.blue-battery.com/produkte?gclid=EAIaIQobChMIuvihioW8gQMVLgWiAx2zuQlTEAAYASAAEgJG7fD_BwE) have some battery computer.
This device is like a Victron SmartShunt.
The BlueBattery+ APP offers a Home Assistant connection.
But a Android Mobile Phone must be placed between the Home Assistant Server and the BlueBattery device.
Daniel Fett has a github [project](https://github.com/danielfett/bluebattery.py) with a Raspberry Pi connection coded in phython.
Is it possible to integrate the code from Daniel in your batmon project?

BR
Andreas

**Comment by fl4p** [comment]
Yes thats possible. I don't own such a device. Happy about a pull request

**Comment by Karatzie** [comment]
Hi Fabian.
I also don't own such device but I know somebody who own that device and connect it over a seperate Raspberry Pi.
My Supervolt Batteries measure the low curent not exactly and the SOC is more a view in a glassball.
I try to calculate the SOC over Home Assistant.
This is also not so correct because the Bluetooth communication is often disturb.

My campervan will be placed in a parkhouse in the winter.
I will check all possibilities and invest in BlueBattery or Victron SmartShut next year.

ER
Andreas

**Reply by fl4p** [reply]
I am designing an open source Smart shunt similar to the Victron and BlueBattery. It will be based on a ESP32 and the INA226. Parts will be less than 20€ and it'll have Wifi and BLE.

If you're interested I can send you a device for testing when I have one.

**Comment by Karatzie** [comment]
That sounds good.
I am interested for your solution.
But I can check the device after the 1. Aprilof the next year.

BR
Andreas

**Comment by Karatzie** [comment]
Hi Fabian.

I have ordered two INA226 and two 100 A shunt. :)
I have an ESP32 and I want to use it for additional tasks.
I want to use a PWM output for a PC fan.
Additional is it possible to connect the Truma heating over a LIN bus interface.
The Truma interface is already configured over ESPhome.
The PWM output and the INA226 can also be integrate over ESPhome.
I want to try this way but it would be nice if you inform me about your solution.

BR
Andreas

**Comment by Karatzie** [comment]
Hi Fabian,

I found a PDF file with the Bluetooth protocol description of the Metrisun (Meritsun) battery.
The protocol has the same structure like the Supervolt battery.

[bms-metrisun-2.pdf](https://github.com/fl4p/batmon-ha/files/13964571/bms-metrisun-2.pdf)

It is possible, that the Metrisun batteries also running with your batmon-ha.

Additional information: [https://github.com/Jakeler/bms-parser/issues/4](url)

Best regards
Andreas
